src/main/java/yasp/processors/warding/Wards.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package yasp.processors.warding;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+
+import skadistats.clarity.event.Event;
+import skadistats.clarity.event.EventListener;
+import skadistats.clarity.event.Initializer;
+import skadistats.clarity.event.Provides;
+import skadistats.clarity.model.Entity;
+import skadistats.clarity.model.FieldPath;
+import skadistats.clarity.model.CombatLogEntry;
+import skadistats.clarity.processor.entities.OnEntityCreated;
+import skadistats.clarity.processor.entities.OnEntityDeleted;
+import skadistats.clarity.processor.entities.OnEntityUpdated;
+import skadistats.clarity.processor.entities.UsesEntities;
+import skadistats.clarity.processor.gameevents.OnCombatLogEntry;
+import skadistats.clarity.processor.reader.OnTickEnd;
+import skadistats.clarity.processor.runner.Context;
+import skadistats.clarity.wire.common.proto.DotaUserMessages;
+
+/**
+ * @author micaelbergeron
+ */
+@UsesEntities
+@Provides({ OnWardCountered.class, OnWardExpired.class, OnWardPlaced.class })
+public class Wards {
+    
+    private static final Set<String> WARDS_DT_CLASSES = new HashSet<String>(Arrays.asList(
+        new String[] {
+            "CDOTA_NPC_Observer_Ward",
+            "CDOTA_NPC_Observer_Ward_TrueSight"
+        }
+    ));
+    
+    private static final Set<String> WARDS_TARGET_NAMES = new HashSet<String>(Arrays.asList(
+        new String[] {
+            "npc_dota_observer_wards",
+            "npc_dota_sentry_wards"
+        }
+    ));
+    
+    private final Map<Integer, FieldPath> lifeStatePaths = new HashMap<>();
+    private final Map<Integer, Integer> currentLifeState = new HashMap<>();
+
+    private final Queue<String> wardKillers = new ArrayDeque<>();
+    private Queue<ProcessEntityCommand> toProcess = new ArrayDeque<>();
+
+    private Event<OnWardCountered> evCountered;
+    private Event<OnWardExpired> evExpired;
+    private Event<OnWardPlaced> evPlaced;
+    
+    private class ProcessEntityCommand {
+
+        private Entity entity;
+        private FieldPath fieldPath;
+        
+        public ProcessEntityCommand(Entity e, FieldPath p) {
+            entity = e;
+            fieldPath = p;
+        }
+    } 
+        
+    @Initializer(OnWardCountered.class)
```

Comment:
consider changing name to OnWardKilled since that's a more precise description of what happened (a combat log event indicating a killed ward)